sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -136,11 +136,20 @@ object TypeCoercion {
     case (TimestampType, DateType) => Some(TimestampType)
     case (DateType, TimestampType) => Some(TimestampType)
 
-    // There is no proper decimal type we can pick,
-    // using double type is the best we can do.
-    // See SPARK-22469 for details.
-    case (n: DecimalType, s: StringType) => Some(DoubleType)
-    case (s: StringType, n: DecimalType) => Some(DoubleType)
+    case (l, r) if (l.isInstanceOf[StringType] && r.isInstanceOf[NumericType]) ||
+      (l.isInstanceOf[NumericType] && r.isInstanceOf[StringType]) =>
+      if (conf.ansiEnabled) {
+        throw new ArithmeticException(
+          s"""
+            | Operator does not exist: $l compare $r
```

Comment:
We don't have to follow the error message of pgsql. We can return None here, and the framework would produce proper error message for us.

Reply:
resolved